src/FSharp.Azure.Storage/Table.fs

```diff
@@ -61,6 +61,11 @@ module Table =
 
     let private isRecordType t = FSharpType.IsRecord (t, Reflection.BindingFlags.Public ||| Reflection.BindingFlags.NonPublic)
 
+    let private unionFromString (t: Type) (s:string) =
+        match FSharpType.GetUnionCases t |> Array.filter (fun case -> case.Name = s) with
+        |[|case|] -> FSharpValue.MakeUnion(case,[||])
```

Comment:
That's a good point, I'll see if it can be easily detected in reflection.

Reply:
Done some digging and it turns out its a little ugly to try and check unions that take arguments. Basically, you have to walk the parent type and back down again. _But_ if the union doesn't take a parameter then it's not on the parent type, because 🤷‍♂️!

Here's a snippet that will get the `ctor` info for unions:

```fsharp
open Microsoft.FSharp.Reflection
open System.Reflection

type U =
    | A of string
    | B of int
    | C of string * string
    | D
    
let printCtors s =
    match FSharpType.GetUnionCases typeof<U> |> Array.tryFind (fun c -> c.Name = s) with
    | None -> failwithf "%s is not a valid union case" s
    | Some case ->
        let dt = case.DeclaringType
        let nt = dt.GetNestedType s
        let ctors = nt.GetConstructors(BindingFlags.NonPublic ||| BindingFlags.Instance)
        let ctor = ctors.[0]

        printfn "Parameters for %s" s
        ctor.GetParameters() |> Array.iter (fun p -> printfn "Parameter: %A" p)

printCtors "A"
printCtors "B"
printCtors "C"
printCtors "D"
```

Note - it'll fail on the `printCtors "D"` call, since that doesn't have a `ctor` (in fact, it's not a nested type!).

I _could_ add this and do something funky about `ctor` detection and execution, but it is a bit clunky.

Reply:
Actually, if you take a different approach to it, it's very easy to detect if a union type has fields:

```fsharp
let unionTypeHasFields t =
  FSharpType.GetUnionCases t
  |> Seq.exists (fun u -> u.GetFields () |> Seq.isEmpty |> not)
```

I think it's important that we give useful error messages to our users, especially with this sort of non-type-safe reflection stuff where the compiler does not check correct use of the API. We should bomb if any of the cases have fields, not just the one we're deserializing for, or else someone may think their type is working okay until they try to deserialise some data into a case they hadn't tested before.

Reply:
It's like you've spent more time doing F# reflection than I have (I know the C# reflection API well though 🤣).

I'll add that pre-check in the code and update the PR

Reply:
Turns out that it'll serialise just fine, but it's the deserialisation that's the problem, so I added an explicit check for them on write to avoid a scenario where you write data you're unable to read back out.

I also added a test for writing the data and ensuring that it fails, but I'm unsure how to do the inverse test (or if it'd be possible, to test that you can't read something you couldn't originally write).

Reply:
You could do the inverse (checking that reading what's there into a union type with fields fails) by writing it in via a different type and then trying to read it out.

Reply:
Ah yeah that's true. Do you want that test coverage added?

Reply:
I think it'd be worth it. Thank you for persisting with this! :)

Reply:
There we go, test added!